shadows/framework/src/main/java/org/robolectric/shadows/ShadowCryptoObject.java

```diff
@@ -0,0 +1,19 @@
+package org.robolectric.shadows;
+
+import static android.os.Build.VERSION_CODES.P;
+
+import android.hardware.biometrics.CryptoObject;
+import org.robolectric.annotation.HiddenApi;
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+
+@SuppressWarnings("UnusedDeclaration")
+@Implements(value = CryptoObject.class, isInAndroidSdk = false, minSdk = P)
+public class ShadowCryptoObject {
+
+  @Implementation
+  @HiddenApi
+  public final long getOpId() {
```

Comment:
Make it protected not public.

Reply:
It is public in the real API, wouldn't that break something?

Reply:
Yep. But shadow API is not needed to be public except very special ocassion. If you make this shadow API public, other users can retrieve it directly as they can get `ShadowCryptoObject`, and it increase the maintainance affording to not breaking existing projects when removing or changing API signatures at a new Robolectric version.